test/adaptive_avg_pool/problem.py

```diff
@@ -0,0 +1,177 @@
+"""Test problems for the AdaptiveAvgPool shape checker."""
+from __future__ import annotations
```

Comment:
Need your feedback here: Why is this necessary?

Reply:
Because `class AdaptiveAvgPoolProblem` is referenced before definition in a type hint.
Referenced in line 25: `def make_test_problems(settings: List[Dict[str, Any]]) -> List[AdaptiveAvgPoolProblem]:`
Defined in line 82: `class AdaptiveAvgPoolProblem:`